tests/ignite/metrics/test_ssim.py

```diff
@@ -158,19 +176,19 @@ def test_ssim_device(available_device, available_device2):
     assert ssim._kernel.device.type == excepted_device
 
 
-def test_ssim_variable_batchsize(available_device):
+@pytest.mark.parametrize("shape, ndims", [[(28, 28), 2], [(14, 14, 14), 3]])
+def test_ssim_variable_batchsize(available_device, shape, ndims):
     # Checks https://github.com/pytorch/ignite/issues/2532
     sigma = 1.5
     data_range = 1.0
-    ssim = SSIM(data_range=data_range, sigma=sigma, device=available_device)
+    ssim = SSIM(data_range=data_range, sigma=sigma, ndims=ndims, device=available_device)
 
     y_preds = [
-        torch.rand(12, 3, 28, 28, device=available_device),
-        torch.rand(12, 3, 28, 28, device=available_device),
-        torch.rand(8, 3, 28, 28, device=available_device),
-        torch.rand(16, 3, 28, 28, device=available_device),
-        torch.rand(1, 3, 28, 28, device=available_device),
-        torch.rand(30, 3, 28, 28, device=available_device),
+        torch.rand(12, 3, *shape, device=available_device),
+        torch.rand(8, 3, *shape, device=available_device),
+        torch.rand(7, 3, *shape, device=available_device),
```

Comment:
```suggestion
        torch.rand(16, 3, *shape, device=available_device),
```
let's have one larger 12